sites/partners/src/settings/PreferenceDrawer.tsx

```diff
@@ -276,7 +539,45 @@ const PreferenceDrawer = ({ drawerOpen, setDrawerOpen }: PreferenceDrawerProps)
           className={"mt-4"}
           styleType={AppearanceStyleType.primary}
           size={AppearanceSizeType.normal}
-          onClick={() => {
+          onClick={async () => {
+            const formData = getValues() as OptionForm
+            await trigger()
+            if (!formData.optionTitle || formData.optionTitle === "") {
+              setError("optionTitle", { message: t("errors.requiredFieldError") })
+              return
+            }
+            const existingOptionData = questionData?.options?.find(
+              (option) => optionData?.ordinal === option.ordinal
+            )
+
+            const getNewOrdinal = () => {
+              if (existingOptionData) return existingOptionData.ordinal
+              return questionData?.options?.length ? questionData?.options.length + 1 : 1
```

Comment:
This isn't necessarily more readable, but you should be able to get rid of the ternary operator with 
```
return  (questionData?.options?.length || 0) + 1